### PR TITLE
test: Fix intermittent issue in wallet_bumpfee

### PR DESCRIPTION
**Pull request by maflcko**
With a "dirty" mempool a transaction might fail to be accepted intermittently. For example,

* https://travis-ci.org/github/bitcoin-core/gui/jobs/719916499#L6773 Fails acceptance
* https://travis-ci.org/github/bitcoin-core/gui/jobs/719916499#L6954 Test fails

Fix the issue by clearing the mempool between subtests

### CODE REVIEW

File: test/functional/wallet_bumpfee.py [hunk 1]
```diff
@@ -333,6 +358,7 @@ def test_maxtxfee_fails(self, rbf_node, dest_address):
     assert_raises_rpc_error(-4, "Unable to create transaction. Fee exceeds maximum configured by -maxtxfee", rbf_node.bumpfee, rbfid)
     self.restart_node(1, self.extra_args[1])
     rbf_node.walletpassphrase(WALLET_PASSPHRASE, WALLET_PASSPHRASE_TIMEOUT)
+    self.connect_nodes(1, 0)
```
Comment (promag) [inline test/functional/wallet_bumpfee.py:362]: I think I'd prefer to see `clear_mempool()` here, on each test end.

**Review by promag — COMMENTED** [review]
ACK

File: test/functional/wallet_bumpfee.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/wallet_bumpfee.py:356]: thx, done

File: test/functional/wallet_bumpfee.py [hunk 2]
```diff
@@ -211,14 +211,15 @@ def test_notmine_bumpfee_fails(self, rbf_node, peer_node, dest_address):
     # here, the rbftx has a peer_node coin and then adds a rbf_node input
     # Note that this test depends upon the RPC code checking input ownership prior to change outputs
     # (since it can't use fundrawtransaction, it lacks a proper change output)
-    utxos = [node.listunspent()[-1] for node in (rbf_node, peer_node)]
+    fee = Decimal("0.001")
+    utxos = [node.listunspent(query_options={'minimumAmount': fee})[-1] for node in (rbf_node, peer_node)]
```
Comment (promag) [inline test/functional/wallet_bumpfee.py:226]: fa347b2f2570a9436313a7c5d61015f2088662f4

nit, maximumCount: 1.

File: test/functional/wallet_bumpfee.py [hunk 3]
```diff
@@ -211,14 +211,15 @@ def test_notmine_bumpfee_fails(self, rbf_node, peer_node, dest_address):
...
     inputs = [{
         "txid": utxo["txid"],
         "vout": utxo["vout"],
         "address": utxo["address"],
         "sequence": BIP125_SEQUENCE_NUMBER
     } for utxo in utxos]
-    output_val = sum(utxo["amount"] for utxo in utxos) - Decimal("0.001")
+    output_val = sum(utxo["amount"] for utxo in utxos) - fee
     rawtx = rbf_node.createrawtransaction(inputs, {dest_address: output_val})
```
Comment (promag) [inline test/functional/wallet_bumpfee.py:234]: fa347b2f2570a9436313a7c5d61015f2088662f4

nit also assert output_val > 0? `utxo` list "could" be empty.

**Review by promag — COMMENTED** [review]
Code review ACK fafc9d5af48c987212af4170bb8b2504be8138f8.

File: test/functional/wallet_bumpfee.py (hunk 3 above)
Comment (MarcoFalke) [reply test/functional/wallet_bumpfee.py:234]: `createrawtransaction` asserts that the value is positive

**Comment by MarcoFalke** [comment]
Another failure: https://travis-ci.org/github/bitcoin/bitcoin/jobs/721362104#L6432

**Pull request merged**